frontend/pages/MDMAppleSSOCallbackPage/MDMAppleSSOCallbackPage.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+
+import endpoints from "utilities/endpoints";
+
+import Spinner from "components/Spinner/Spinner";
+import SSOError from "components/MDM/SSOError";
+import Button from "components/buttons/Button";
+
+const baseClass = "mdm-apple-sso-callback-page";
+
+const RedirectTo = ({ url }: { url: string }) => {
+  window.location.href = url;
+  return <Spinner />;
+};
+
+// appConfig.ServerSettings.ServerURL+"/api/latest/fleet/mdm/apple/setup/eula/"
+// /api/mdm/apple/enroll?token=
+
+const EnrollmentGate = () => {
+  const query = new URLSearchParams(location.search);
+  const [showEULA, setShowEULA] = useState(query.has("eula_token"));
+  const profileToken = query.get("profile_token");
+  const eulaToken = query.get("eula_token") || "";
```

Comment:
is it correct that you are getting query params here? If so you should have access to them in the props under `location.query` from react router. will this work for getting these values instead of using native APIs?

Reply:
yes, we need to improve our typing around react router.